Himanshu_Kumar/project1-(calculator)/assets/js/cal_js.js

```diff
@@ -0,0 +1,138 @@
+const calculator = {
+  displayValue: '0',
+  firstOperand: null,
+  waitingForSecondOperand: false,
+  operator: null,
+};
+
+function fact(val){
+  let v=1;
+  for(let i=1; i<=val; i++){
+    v=v*i;
+  }
+  return v;
+}
+
+function inputDigit(digit) {
+  const { displayValue, waitingForSecondOperand } = calculator;// declaring and initialising multiple variables in one line
+
+  if (waitingForSecondOperand === true) {
+    calculator.displayValue = digit;
+    calculator.waitingForSecondOperand = false;
+  } else {
+    calculator.displayValue = displayValue === '0' ? digit : displayValue + digit;
+  }
+}
+
+function inputDecimal(dot) {
+  if (!calculator.displayValue.includes(dot)) {
+    calculator.displayValue += dot;// if a dot is already not included, include it on pressing the dot button
+  }
+}
+
+function handleOperator(nextOperator) {
+  const { firstOperand, displayValue, operator } = calculator
+  const inputValue = parseFloat(displayValue);
```

Comment:
@Hhumour 
remember whenever we are making us of parsing, it should be used within try-catch block